lib/event.c

```diff
@@ -886,7 +1200,29 @@ static int fd_poll(struct event_loop *m, const struct timeval *timer_wait,
 		pthread_sigmask(SIG_SETMASK, NULL, &origsigs);
 	}
 
-#if defined(HAVE_PPOLL)
+#if defined(USE_EPOLL) && defined(HAVE_EPOLL_PWAIT2)
+	struct timespec ts, *tsp;
+
+	if (timeout >= 0) {
+		ts.tv_sec = timeout / 1000;
+		ts.tv_nsec = (timeout % 1000) * 1000000;
+		tsp = &ts;
```

Comment:
While slightly outside the scope of this PR, I would appreciate if you could add this as a new helper to `monotime.h`, like `ms2ts()` or something, and then call that here.